### ISSUE TITLE
`vue/no-v-html` should not fail when `TrustedHTML` is passed

### ISSUE THREAD

**Issue by shaedrich** [vuejs/eslint-plugin-vue#3078]
**What rule do you want to change?** `vue/no-v-html`

**Does this change cause the rule to produce more or fewer warnings?** fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
I would like this to be the default behavior but for backwards compatibility, I'd be fine with it being an option (and maybe in a later version made the default)

**Please provide some example code that this change will affect:**

```vue
<template>
    <!-- should report -->
    <span v-html="rawHtml"></span>
    <!-- should NOT report -->
    <span v-html="escapedHtml"></span>
</template>

<script setup>
const rawHtml = '<img src=x onerror=alert(1)>'
const escapeHTMLPolicy = trustedTypes.createPolicy("myEscapePolicy", {
  createHTML: (string) => string.replace(/</g, '&lt;').replace(/>/g, '&gt;'),
})

const escapedHtml = escapeHTMLPolicy.createHTML(rawHtml)
</script>
```

**What does the rule currently do for this code?**
Reports in both cases

**What will the rule do after it's changed?**
Report only in the first case

**Additional context**
* https://developer.mozilla.org/en-US/docs/Web/API/TrustedHTML

**Comment by waynzh** [comment]
Maybe we could skip the check if we have a TS parser in place by using TypeScript types to determine if it's a `TrustedHTML` type. What do you think?

**Comment by shaedrich** [comment]
You tell me 😂 Sounds plausible 👍🏻

**Comment by ota-meshi** [comment]
I suspect TypeScript doesn't have those types yet.
Therefore, checking them using types might still be difficult.
https://github.com/microsoft/TypeScript/issues/30024
